### PR TITLE
refactor(eslint-plugin): use latest typescript versions in peerDependencies

### PR DESCRIPTION
**Pull request by mimse**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## What is the new behavior?

It is fixing a yarn dependency error

image

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by brandonroberts** [comment]
Thanks @mimse. This seems like a yarn issue, because `>=4.4.0` should be inclusive of `5.0.0` also

File: modules/eslint-plugin/package.json [hunk 1]
```diff
@@ -44,6 +44,6 @@
   },
   "peerDependencies": {
     "eslint": ">=8.0.0",
-    "typescript": ">=4.4.0"
+    "typescript": ">=4.4.0 || >=5.0.0"
```
Comment (timdeschryver) [inline modules/eslint-plugin/package.json:47]: Let's use a `*` so we don't bump into this anymore.
It seems like the angular eslint project is also using this https://github.com/angular-eslint/angular-eslint/blob/main/packages/eslint-plugin/package.json#L25C1-L25C22

```suggestion
    "typescript": "*"
```

**Review by timdeschryver — CHANGES_REQUESTED** [review]

File: modules/eslint-plugin/package.json (hunk 1 above)
Comment (mimse) [reply modules/eslint-plugin/package.json:47]: Praise: Good solution, no need to worry about it again

**Review by brandonroberts — APPROVED** [review]

**Review by timdeschryver — APPROVED** [review]
Thanks @mimse !

**Pull request merged**